src/OpenTelemetry.AutoInstrumentation/Instrumentations/Wcf/ServiceHostIntegration.cs

```diff
@@ -0,0 +1,58 @@
+// <copyright file="ServiceHostIntegration.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+#if NETFRAMEWORK
+using OpenTelemetry.AutoInstrumentation.CallTarget;
+using OpenTelemetry.AutoInstrumentation.Configurations;
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.Wcf;
+
+/// <summary>
+/// ServerHostIntegration.
+/// </summary>
+[InstrumentMethod(
+    assemblyName: WcfCommonConstants.ServiceModelAssemblyName,
+    typeName: WcfServiceConstants.ServiceHostBaseTypeName,
+    methodName: WcfServiceConstants.InitializeDescriptionMethodName,
+    returnTypeName: ClrNames.Void,
+    parameterTypeNames: new[] { WcfServiceConstants.UriSchemeKeyedCollectionTypeName },
+    minimumVersion: WcfCommonConstants.MinVersion,
+    maximumVersion: WcfCommonConstants.MaxVersion,
+    integrationName: WcfServiceConstants.IntegrationName,
+    type: InstrumentationType.Trace)]
+public static class ServiceHostIntegration
+{
+    /// <summary>
+    /// OnMethodEnd callback
+    /// </summary>
+    /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+    /// <param name="exception">Exception value</param>
+    /// <param name="state">CallTarget state</param>
+    /// <typeparam name="TTarget">Type of the target</typeparam>
+    /// <returns>A response value, in an async scenario will be T of Task of T</returns>
+    internal static CallTargetReturn OnMethodEnd<TTarget>(TTarget instance, Exception exception, in CallTargetState state)
```

Comment:
Since you know the instance type, you can use the following generic constraint to immediately use the DuckType interface without having to do the `TryDuckCast` call later. This isn't required, but can make the code easier to follow since you're not having to do an if/else later

```suggestion
    internal static CallTargetReturn OnMethodEnd<TTarget>(TTarget instance, Exception exception, in CallTargetState state)
        where TTarget : IServiceHostBase
```